### ISSUE TITLE
Laggy Scrolling in Viewer

### ISSUE THREAD

**Issue by ProfessorCode212** [laurent22/joplin#4827]
Scrolling through the Markdown viewer is quite laggy and not as smooth as it should be.

https://user-images.githubusercontent.com/78950242/114266474-31c5c480-9a14-11eb-876c-4a397f3f2f3d.mp4

## Environment

Joplin version: 1.7.11
Platform: Linux
OS specifics: Linux Mint 19.3, AppImage

## Steps to reproduce

1. Open a note. 
2. Scroll through the note with the _Viewer_.

## Describe what you expected to happen
Scrolling should be smooth and seamless without 'breaking' in the middle.

**Comment by tannercollin** [comment]
I'm also having this issue in the full markdown viewer (not only side-by-side). Unfortunately the bug makes this view pretty useless.


https://user-images.githubusercontent.com/6752449/118777878-c8bd4200-b846-11eb-87bb-f54340f0642a.mp4

**Comment by ProfessorCode212** [comment]
@tannercollin It's the same for me as well, although I only showed it using _Split View_.

**Comment by ProfessorCode212** [comment]
I updated to version 2.0.11 and the problem still exists.

**Comment by qrsp** [comment]
I have the same issue when zooming out the layout, but scrolling smoothly on `Actual Size`

**Comment by tannercollin** [comment]
Yup, still a problem on v2.1.7 but only when zoomed in or out. Default zoom (Ctrl+0) fixes the issue, but then everything's too small to read. :eyeglasses:

**Comment by tannercollin** [comment]
Still a problem on v2.1.9.

**Comment by tannercollin** [comment]
@laurent22 why was the "bug" label removed?

**Comment by diikhub** [comment]
I had this scrolling issue since I have started using Joplin. Scrolling becomes laggy at will. I wish the so called "smooth" scrooling could be disabled. You know the old fashioned "jumpy" scrolling which scrolls just 3 lines or so. It's faster as well. Firefox still has the option to turn off the smooth scrolling.

Joplin is the only application I am unable to "force" not to  use smooth scrolling. Too bad, I love Joplin otherwise :-)

**Comment by ken1kob** [comment]
I analyzed this issue and found a workaround.

As the above reports said, this symptom happens in the Markdown viewer when zoomFactor is not 1. So, users of HD displays may often suffer this symptom.

The following is an example of debug log of this symptom collected when displaying "Welcome to Joplin!" note with 1-level zoom out. To collect this log, I scrolled down the note slowly and tried several mice to confirm there is no hardware malfunctions or operational errors. Each line shows the state of `joplin-container-content` element for each scroll event.

```
note-viewer/index.html: scrollH=2970 clientH=687 scrollTop=277.77777099609375
note-viewer/index.html: scrollH=2970 clientH=687 scrollTop=276.66668701171875
note-viewer/index.html: scrollH=2970 clientH=687 scrollTop=277.77777099609375
note-viewer/index.html: scrollH=2970 clientH=687 scrollTop=276.66668701171875
note-viewer/index.html: scrollH=2970 clientH=687 scrollTop=278.8888854980469
note-viewer/index.html: scrollH=2970 clientH=687 scrollTop=277.77777099609375
note-viewer/index.html: scrollH=2970 clientH=687 scrollTop=276.66668701171875
note-viewer/index.html: scrollH=2970 clientH=687 scrollTop=277.77777099609375
note-viewer/index.html: scrollH=2970 clientH=687 scrollTop=276.66668701171875
```

As you can see, even though I were constantly scrolling down, the values of `scrollTop` goes up and down and cycles.

Since DOM element's  `scrollTop` is calculated by the upstream (Electron or Chromium) and they are implemented by C++, Joplin cannot interere. So, the symptom is not caused by Joplin.

Such troubles like this symptom have been frequently reported in these years after features about subpixel/zoomFactor are introduced. You can see [many open/closed bugs of Chromium here](https://bugs.chromium.org/p/chromium/issues/list?q=scrolltop%20zoom&can=2).


A workaround is not difficult. It is enough to calculate `scrollTop` by yourself. It is the reason the Markdown Editor (CodeMirror), the WYSIWYG editor (TinyMCE) and other popular products  are free from this symptom.

Soon, I'll send a pull request fixes the issue.
